system/jlib/jptree.cpp

```diff
@@ -10130,3 +10130,88 @@ void setExpertOpt(const char *opt, const char *value)
     getExpertOptPath(opt, xpath.clear());
     config->setProp(xpath, value);
 }
+
+//---------------------------------------------------------------------------------------------------------------------
+
+//More: Should this move inside PTree?  It may allow a more efficient hash caclulaton.
+unsigned getPropertyTreeHash(const IPropertyTree & source, unsigned hashcode)
```

Comment:
Feels like this will be an expensive calculation on a large tree, that would be avoided if you could store hash of each component when it was created. But maybe it's not going to be used on large trees?

Reply:
Created HPCC-30752 to address optimizing it.  Currently it will be used on small trees.